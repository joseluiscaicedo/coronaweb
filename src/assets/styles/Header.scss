.header {
  align-items: center;
  display: flex;
  height: 100px;
  justify-content: space-between;
  top: 0px;
  width: 100%;
}

.header__img {
  margin-left: 10px;
  width: 200px;
}

.header__menu {
	margin-right: 30px;
}

.header__menu ul {
  display: none;
  list-style: none;
  margin: 0px 0px 0px -14px;
  padding: 0px;
  position: absolute;
  width: 100px;
  text-align: right;
}

.header__menu:hover ul, ul:hover {
	display: block;
}

.header__menu li {
  margin: 10px 0px;
}

.header__menu li a {
  text-decoration: none;
}

.header__menu li a:hover {
  text-decoration: underline;
}

.header__menu--profile {
  align-items: center;
  display: flex;
  cursor: pointer;
}

.header__menu--profile img {
  margin-right: 8px;
  width: 40px;
}