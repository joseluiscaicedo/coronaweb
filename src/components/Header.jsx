import React from 'react';
import { useAuth0 } from '@auth0/auth0-react';

import LoginButton from './LoginButton';
import LogoutButton from './LogoutButton';
import Profile from './Profile';

import logo from '../assets/images/logo.png';
import userIcon from '../assets/images/user-icon.png';

import '../assets/styles/Header.scss';

const Header = () => {
  const { isAuthenticated } = useAuth0();
  return (
    <header aria-label='header title' className='header'>
      <img className='header__img' src={logo} alt='coronaweb logo' />
      <div className='header__menu'>
        {isAuthenticated ? (
          <>
            <Profile />
            <LogoutButton />
            {' '}
          </>
        ) :
          (
            <>
              <div className='header__menu'>
                <img id='menu__img' src={userIcon} alt='user Icon' />
                <LoginButton />
              </div>
            </>
          )}
      </div>
    </header>
  );
};

export default Header;
