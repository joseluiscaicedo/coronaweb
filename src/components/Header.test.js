import React from 'react';
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import Header from './Header'

describe('validate Header', () =>{
  const header = render(<Header />)
  test('renders content', () => {

    expect(header.length)>0;

  });

  test('Generates Snapshot for COMPONENT Component', () => {
    const { container } = render(
      <Header />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

})