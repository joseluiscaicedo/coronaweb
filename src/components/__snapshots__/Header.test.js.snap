// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate Header Generates Snapshot for COMPONENT Component 1`] = `
<header
  aria-label="header title"
  class="header"
>
  <img
    alt="coronaweb logo"
    class="header__img"
    src="[object Object]"
  />
  <div
    class="header__menu"
  >
    <div
      class="header__menu"
    >
      <img
        alt="user Icon"
        id="menu__img"
        src="[object Object]"
      />
      <button
        class="ButtonLogin"
        id="ButtonLogin"
        type="button"
      >
        Login
      </button>
    </div>
  </div>
</header>
`;
